Echo/EchoNode.cpp

```diff
@@ -0,0 +1,246 @@
+#include "EchoNode.hpp"
+#include <iostream>
+#include <fstream>
+
+
+using namespace std;
+
+
+namespace polysync
+{
+
+
+void PolySyncEcho::initStateEvent()
+{
+    if ( _filteredForSingleMsgFlag )
+    {
+        try
+        {
+            registerListener( getMessageTypeByName ( _msgName ) );
+        }
+
+        catch ( ... )
+        {
+            std::cout << "\nPlease enter a valid PS message type. \n";
+
+            printAvailableMessage( getAvailableMessageNames() );
+
+            disconnectPolySync();
+        }
+    }
+
+    else
+    {
+        registerListenerToAllMessageTypes();
+    }
+}
+
+
+void PolySyncEcho::setConfigurationEvent( int argc, char * argv [] )
+{
+    std::string filterFlag = "-filterMessage";
+    std::string echoHeadersOnly = "-echoHeaders";
```

Comment:
Maybe there should be a check for invalid command line arguments? So if they enter a flag that doesn't make sense there will be an error that maybe lets them know they can use the help flag? It looks like for now if there is an unexpected flag, like if I misspell "-filterMessage" it just gets ignored without me knowing it didn't take.